includes/class-wp-job-manager-com-api.php

```diff
@@ -0,0 +1,172 @@
+		$response = $this->request(
+			'wpjmcom-notices/1.0/notices',
+			[
+				'version' => JOB_MANAGER_VERSION,
+				'lang'    => determine_locale(),
+			],
+			DAY_IN_SECONDS,
+			$max_age
+		);
+
+		if ( is_wp_error( $response ) || ! is_array( $response ) ) {
+			return [];
+		}
+
+		return $response['notices'];
+	}
+
+	/**
+	 * Make a GET request to the API and caches content.
+	 *
+	 * @param string   $path      The request path.
+	 * @param array    $args      The request GET parameters.
+	 * @param int      $cache_ttl The cache TTL.
+	 * @param int|null $max_age   The max age (seconds) of the source data. If older than this the data will be fetched again.
+	 * @return array|\WP_Error
+	 */
+	protected function request( $path, $args, $cache_ttl = DAY_IN_SECONDS, $max_age = null ) {
+		$transient_key = $this->get_request_transient_key( $path, $args );
+		$cached_data   = get_transient( $transient_key );
+
+		// If the cached data is too old, ignore it.
+		if ( $max_age && is_array( $cached_data ) ) {
+			$age = time() - ( $cached_data['_fetched'] ?? 0 );
+			if ( $age > $max_age ) {
+				$cached_data = false;
+			}
+		}
+
+		// If no cached data or malformed, fetch it.
+		if ( false === $cached_data || ! is_array( $cached_data ) || ! isset( $cached_data['_remote_data'] ) ) {
+			$response = wp_safe_remote_get(
+				add_query_arg(
+					$args,
+					$this->get_api_base_url() . $path
+				),
+				[
+					'timeout' => 10,
+					'headers' => [
+						'Accept'       => 'application/json',
+						'Content-Type' => 'application/json',
+					],
+				]
+			);
+
+			// If WP_Error return without caching.
+			if ( is_wp_error( $response ) ) {
+				return $response;
+			}
+
+			// If HTTP error return without caching.
+			if ( 200 !== wp_remote_retrieve_response_code( $response ) ) {
+				$error_code = wp_remote_retrieve_response_code( $response );
+				return new \WP_Error( $error_code, 'Error code: ' . $error_code );
+			}
+
+			$remote_data = json_decode( wp_remote_retrieve_body( $response ), true );
+
+			if ( ! is_null( $remote_data ) ) {
+				$cached_data = [
+					'_fetched'     => time(),
+					'_remote_data' => $remote_data,
+				];
+				set_transient( $transient_key, $cached_data, $cache_ttl );
+			}
+
+			return $remote_data;
+		}
+
+		return $cached_data['_remote_data'];
+	}
+
+	/**
+	 * Returns the API base URL.
+	 *
+	 * @return string
+	 */
+	private function get_api_base_url() {
+		// For backwards compatibility we check and use the JOB_MANAGER_DEV_API_BASE_URL constant.
+		if (
+			defined( 'JOB_MANAGER_VERSION' )
+			&& defined( 'JOB_MANAGER_DEV_API_BASE_URL' )
+			&& '-dev' === substr( JOB_MANAGER_VERSION, -4 )
+		) {
+			return JOB_MANAGER_DEV_API_BASE_URL;
+		}
+
+		/**
+		 * Filters the wpjobmanager.com API URL.
+		 *
+		 * @hook   wpjmcom_api_url
+		 *
+		 * @param  {string}   $api_url The API url.
+		 *
+		 * @return {array} The API url.
+		 */
+		return apply_filters( 'wpjmcom_api_url', self::API_BASE_URL );
```

Comment:
True. Renamed in 8d4489b9

I am using `wpjm_*` since it's the prefix most recently used. I have asked in slack and will adapt if Zen suggests any other preference.